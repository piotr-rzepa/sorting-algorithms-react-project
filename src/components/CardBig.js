/*
 * Komponent bezklasowy -> wariacja karty, z innym stylem
 */

/* eslint-disable react/prop-types */
import React from 'react';

export default function CardBig({ desc }) {
	return (
		<div className="Card Card--big">
			<h4>{desc}</h4>
		</div>
	);
}
