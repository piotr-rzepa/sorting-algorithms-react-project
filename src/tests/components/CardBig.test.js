import * as React from 'react';
import { shallow } from 'enzyme';
import CardBig from '../../components/CardBig';

const data = {
	title: 'main-title-test',
	subtitle: 'subtitle-test',
	icon: undefined,
	background: { color: 'white' }
};

describe('<Card />', () => {
	const useContextMock = jest.spyOn(React, 'useContext');
	beforeEach(() => {
		useContextMock.mockClear();
	});
	it('renders correctly', () => {
		const wrapper = shallow(<CardBig props={data} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should choose correct theme', () => {
		useContextMock.mockReturnValue(true);
		const wrapper = shallow(<CardBig props={data} />);
		expect(wrapper.find('div.Card').get(0).props.style).toHaveProperty(
			'color',
			'white'
		);
	});
});
