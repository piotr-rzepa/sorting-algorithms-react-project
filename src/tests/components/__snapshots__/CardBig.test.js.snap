// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<div
  className="Card Card--big"
  style={
    Object {
      "backgroundColor": "white",
      "boxShadow": "2px 2px 2px lightgray",
      "color": "black",
    }
  }
>
  <div
    className="svg-image-background"
  >
    <img
      alt="Icon"
      height={38}
      width={38}
    />
  </div>
  <div
    className="Card-description"
  >
    <h4 />
  </div>
</div>
`;
